cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceRewriter.java

```diff
@@ -350,18 +408,28 @@ protected void onMethodEnter() {
 
           // push the parameters of method call on to the stack
 
-          // pushed the entity id
-          //TODO: This is because currently we don't support multi-part
-          EntityPartDetail entityPart = annotationDetail.getEntities().get(0);
-          if (entityPart.isField()) {
-            // load class field
-            loadThis();
-            getField(classType, entityPart.getEntityName(), fieldDetails.get(entityPart.getEntityName()));
-          } else {
-            // load method parameter
-            loadArg(0);
+          // push the entity array
+          push(annotationDetail.getEntities().size());
```

Comment:
So this will create a new array for every call. Maybe better to have the old logic if the `EntityId` is already there and have a corresponding method in the `AuthEnforceUtil` to support that.